worker/uniter/uniter.go

```diff
@@ -683,6 +679,10 @@ func (u *Uniter) init(unitTag names.UnitTag) (err error) {
 	}
 	u.operationExecutor = operationExecutor
 
+	// Ensure we have an agent directory to to write the socket.
+	if err := os.MkdirAll(u.paths.State.BaseDir, 0755); err != nil {
```

Comment:
We may be able to move this to the uniter tests, as in the real world it appears that it's not needed.  More research necessary.  Will investigate as a refinement.